build/includes/terraform.mk

```diff
@@ -16,54 +16,59 @@
 terraform-init:
 terraform-init: $(ensure-build-image)
 	docker run --rm -it $(common_mounts) $(DOCKER_RUN_ARGS) $(build_tag) bash -c '\
-	cd $(mount_path)/install/terraform && terraform init && gcloud auth application-default login'
+	cd $(mount_path)/examples/terraform-submodules/gke-local && terraform init && gcloud auth application-default login'
 
 terraform-clean:
-	rm -r ../install/terraform/.terraform
-	rm ../install/terraform/terraform.tfstate*
+	rm -r ../examples/terraform-submodules/gke-local/.terraform || true
+	rm ../examples/terraform-submodules/gke-local/terraform.tfstate* || true
 
 # Creates a cluster and install release version of Agones controller
 # Version could be specified by AGONES_VERSION
 gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT ?= 4
 gcloud-terraform-cluster: GCP_CLUSTER_NODEPOOL_MACHINETYPE ?= n1-standard-4
 gcloud-terraform-cluster: AGONES_VERSION ?= ''
+gcloud-terraform-cluster: GCP_CLUSTER_NAME ?= test-cluster
 gcloud-terraform-cluster: $(ensure-build-image)
 gcloud-terraform-cluster:
 ifndef GCP_PROJECT
 	$(eval GCP_PROJECT=$(shell sh -c "gcloud config get-value project 2> /dev/null"))
 endif
-	$(DOCKER_RUN) bash -c 'export TF_VAR_agones_version=$(AGONES_VERSION) && \
-		cd $(mount_path)/install/terraform && terraform apply -auto-approve -var values_file="" \
-		-var chart="agones" \
-	 	-var "cluster={name=\"$(GCP_CLUSTER_NAME)\", machineType=\"$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)\", \
-		 zone=\"$(GCP_CLUSTER_ZONE)\", project=\"$(GCP_PROJECT)\", \
-		 initialNodeCount=\"$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)\"}"'
-	$(MAKE) gcloud-auth-cluster
+	$(DOCKER_RUN) bash -c 'cd $(mount_path)/examples/terraform-submodules/gke-local && \
+		 terraform apply -auto-approve  -var agones_version="$(AGONES_VERSION)" \
+		-var name=$(GCP_CLUSTER_NAME) -var machine_type="$(GCP_CLUSTER_NODEPOOL_MACHINETYPE)" \
+		-var values_file="" \
+		-var zone="$(GCP_CLUSTER_ZONE)" -var project="$(GCP_PROJECT)" \
+		-var node_count=$(GCP_CLUSTER_NODEPOOL_INITIALNODECOUNT)'
+	GCP_CLUSTER_NAME=$(GCP_CLUSTER_NAME) $(MAKE) gcloud-auth-cluster
 
 # Creates a cluster and install current version of Agones controller
 # Set all necessary variables as `make install` does
+# Unifies previous `make gcloud-test-cluster` and `make install` targets
```

Comment:
Should we delete at least `make gcloud-test-cluster` and/or `make install` ? Or rename all the terraform scripts into those targets?

Moving all the developer tooling over to terraform has some really nice implications for e2e testing as well - since we could update Our kubernetes version with a change to a terraform script, and CI would apply the change, which is awesome.

(oooh, that could also be bad, don't want competing CI builds adding and removing versions. Maybe we have to name clusters based on version, or something).

Doesn't have to be in this PR, but figured I would ask.

Reply:
That question need more investigation, by the way Terraform providers looks less stable to me, than what we have now for development tooling

Reply:
Sounds like we will have two ways of developing - one with the terraform script, and then also this usual way, and we can test the terraform way until we feel comfortable with it being switched over?

Works for me.

Reply:
Yes, and also currently there is no make target for `make uninstall && make install`, which was pretty fast and eliminate creating cluster every time to aplly some fix